low-level/files/opt/seagate/sspl/setup/resource_map/server.py

```diff
@@ -32,18 +44,125 @@ class ServerMap(ResourceMap):
+        utility_instance = ToolFactory().get_instance('procfs')
+        mem_path = utility_instance.get_proc_memory('meminfo')
+        if mem_path.is_file():
+            meminfo_dict = self.read_mem_file(mem_path)
+            total_mem = meminfo_dict['MemTotal:']
+            path_exists, _ = self.store.exists(self.MEM_FAULT_SENSOR_DATA)
+            if path_exists:
+                self.get_stored_mem_info()
+                if self.fault_alert_state == "fault":
+                    status = "FAULT"
+                    description  = \
+                    "Total available main memory value decreased"\
+                    f"from {self.prev_mem} kB to {total_mem} kB"
+            memory_dict = self.prepare_mem_json(status,
+                                                description)
+            data.append(memory_dict)
+        else:
+            raise HealthProviderError(
+                errno.EINVAL,
+                f"{mem_path}: file does not exist")
+        return data
+
+    def read_mem_file(self, mem_path):
+        """ Get Memory information from /proc/meminfo."""
+        meminfo_dict = {}
+        self.mem_path_file = mem_path.read_text()
+        # populate all the data from /proc/meminfo
+        split_strs = [s.split(maxsplit=1) for s in self.mem_path_file.splitlines()]
+        meminfo_dict = dict(split_strs)
+        return meminfo_dict
+
+    def get_stored_mem_info(self):
+        """ Get the memory info from filestore """
+        mem_data = (self.store.get(self.MEM_FAULT_SENSOR_DATA)).split(":")
+        self.prev_mem = mem_data[0].strip()
+        self.fault_alert_state = mem_data[1].strip()
+
+    def prepare_mem_json(self, status, description):
+        """Update and return memory information dict """
+        values = dict(psutil.virtual_memory()._asdict())
+        total_memory = {}
+        for key, value in values.items():
+            if key == 'percent':
+                total_memory['percent'] = str(values['percent']) + '%'
+            else:
+                total_memory[key] = str(values[key] >> 20) + 'MB'
 
+        memory_dict = {
+            "uid": "memory_0",
+            "fru": "false",
+            "last_updated": int(time.time()),
+            "health": {
+            "status": status,
+            "description": description,
+            "specifics": [
+                {
+                   "total": total_memory['total'],
+                   "available": total_memory['available'],
+                   "percent": total_memory['percent'],
+                   "used": total_memory['used'],
+                   "free": total_memory['free'],
+                   "active": total_memory['active'],
+                   "inactive": total_memory['inactive'],
+                   "buffers": total_memory['buffers'],
+                   "cached": total_memory['cached'],
+                   "shared": total_memory['shared'],
+                   "slab": total_memory['slab']
+                }
```

Comment:
Need DIMM instances too here, with some health specifics per DIMM. check memory fault sensor, we can reuse code from there for getting the memory fault, which uses sysfs, explore that for churning out more health info